### ISSUE TITLE
//usr/lib/libpq.so.5 errors occurred when compiled the packages!

### ISSUE THREAD

**Issue by Yanzqing** [strands-project/v4r_ros_wrappers#21]
My ROS version is Indigo!
How can solve this problem? 
//usr/lib/libpq.so.5: undefined reference to `SSL_get_error@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_set_ex_data@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `X509_STORE_set_flags@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSLv23_method@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `CRYPTO_set_id_callback@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_CTX_get_cert_store@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `X509_free@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_free@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_set_fd@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_use_PrivateKey@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_get_peer_certificate@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_use_certificate_file@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `ENGINE_free@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`OPENSSL_config@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_CTX_load_verify_locations@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`ENGINE_init@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `ERR_reason_error_string@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_connect@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_library_init@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_shutdown@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_CTX_use_certificate_chain_file@OPENSSL_1.0.0'
/home/exbot/My_packages/Naoqi_robot/sample_code/V4R/v4r-master/build/lib/libv4r_rendering.so.1.0.0: undefined reference to`XISelectEvents'
//usr/lib/libpq.so.5: undefined reference to `SSL_use_PrivateKey_file@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`X509_get_subject_name@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `ENGINE_load_private_key@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_check_private_key@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `ENGINE_by_id@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`CRYPTO_set_locking_callback@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `X509_NAME_get_text_by_NID@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_write@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_ctrl@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_set_verify@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `ENGINE_finish@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`CRYPTO_num_locks@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `X509_STORE_load_locations@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_read@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_CTX_new@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`ERR_get_error@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_new@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to`SSL_load_error_strings@OPENSSL_1.0.0'
/home/exbot/My_packages/Naoqi_robot/sample_code/V4R/v4r-master/build/lib/libv4r_rendering.so.1.0.0: undefined reference to `XIQueryVersion'
//usr/lib/libpq.so.5: undefined reference to`SSL_CTX_ctrl@OPENSSL_1.0.0'
//usr/lib/libpq.so.5: undefined reference to `SSL_pending@OPENSSL_1.0.0'
collect2: error: ld returned 1 exit status

**Comment by marc-hanheide** [comment]
What version of Ubuntu are you using? 32 or 64 bit? please paste the output of `lsb_release -a` and `uname -a`, and also the exact commands you used to compile the wrappers. Also please provide the output of `dpkg -s ros-indigo-v4r` to see what version you are basing this on.

**Comment by Yanzqing** [comment]
I am using Ubuntu 14.04, 64bit. 
 $ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.3 LTS
Release:    14.04
Codename:   trusty

$ uname -a
Linux dfs-Vostro-270 3.13.0-65-generic #105-Ubuntu SMP Mon Sep 21 18:50:58 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

dpkg -s ros-indigo-v4r
Package: ros-indigo-v4r
Status: install ok installed
Priority: extra
Section: misc
Installed-Size: 12521
Maintainer: Thomas Faeulhammer faeulhammer@acin.tuwien.ac.at
Architecture: amd64
Version: 1.1.1-5trusty-20151123-1549-+0000
Depends: libassimp3 (>= 2.0.863), libboost-filesystem1.54.0, libboost-program-options1.54.0, libboost-regex1.54.0, libboost-serialization1.54.0, libboost-system1.54.0, libboost-thread1.54.0, libc6 (>= 2.15), libceres, libdevil1c2, libgcc1 (>= 1:4.1.1), libgl1-mesa-glx | libgl1, libglew1.10 (>= 1.10.0), libgomp1 (>= 4.4), libgoogle-glog0, libopencv-calib3d2.4, libopencv-core2.4, libopencv-features2d2.4, libopencv-flann2.4, libopencv-highgui2.4, libopencv-imgproc2.4, libopencv-video2.4, libpcl-apps-1.7, libpcl-common-1.7, libpcl-features-1.7, libpcl-filters-1.7, libpcl-io-1.7, libpcl-kdtree-1.7, libpcl-keypoints-1.7, libpcl-octree-1.7, libpcl-recognition-1.7, libpcl-sample-consensus-1.7, libpcl-search-1.7, libpcl-segmentation-1.7, libpcl-surface-1.7, libpcl-visualization-1.7, libqt4-opengl (>= 4:4.5.3), libqtcore4 (>= 4:4.7.0~beta1), libqtgui4 (>= 4:4.6.1), libstdc++6 (>= 4.6), libsvm3, libvtk5.8, libvtk5.8-qt4, libx11-6 (>= 2:1.2.99.901), libxcursor1 (>> 1.1.2), libxinerama1, libxrandr2 (>= 2:1.2.99.3), freeglut3-dev, libassimp-dev, libceres-dev, libdevil-dev, libglew-dev, libglm-dev, libgoogle-glog-dev, libopencv-dev, libopenni2-dev, libpcl-1.7-all-dev, libqt4-dev, libqt4-opengl-dev, libsvm-dev, libx11-dev, libxi-dev, libxinerama-dev, libxrandr-dev
Description: The Vision for Robotics Library of ACIN TUW (open source STRANDS version)

I compiled the wrappers via catkin_make in my workspace.
I also installed the latest ros-indigo-v4r-ros-wrappers and ros-indigo-v4r from synaptic package manager, but when i run
 roslaunch  '/opt/ros/indigo/share/singleview_object_recognizer/launch/recognition_service.launch'
ERROR: cannot launch node of type [singleview_object_recognizer/recognition_service]: can't locate node [recognition_service] in package [singleview_object_recognizer]

**Comment by marc-hanheide** [comment]
Right... I cannot see what's going wrong here, to be honest. When you simply install ros-indigo-v4r-ros-wrappers it _should_ contain recognition_service. I tested on my machine and it seems to work... I tried `roslaunch singleview_object_recognizer recognition_service.launch models_dir:=/tmp training_dir:=/tmp`

I have only the Ubuntu packages installed on that machine. Maybe @ToMadoRe, @mzillich  or @taketwo have ideas why it's not working. 

Just to be sure, can you update your system with `apt-get update` and `apt-get upgrade` once more, please. And make sure you have no own ROS workspace configured, so that indeed only the installed packages are being used.

**Comment by Yanzqing** [comment]
I solve the problem
 cannot launch node of type [singleview_object_recognizer/recognition_service]: can't locate node [recognition_service] in package [singleview_object_recognizer]
Thank you for your help!

**Comment by MRLoghmani** [comment]
@Yanzqing could you please report how you solved the initial problem?
